jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java

```diff
@@ -72,11 +69,13 @@ public class OnHeapGraphIndex implements GraphIndex {
     private final AtomicInteger maxNodeId = new AtomicInteger(-1);
 
     // Maximum number of neighbors (edges) per node per layer
-    final IntArrayList maxDegrees;
+    final List<Integer> maxDegrees;
     // The ratio by which we can overflow the neighborhood of a node during construction. Since it is a multiplicative
     // ratio, i.e., the maximum allowable degree if maxDegree * overflowRatio, it should be higher than 1.
     private final double overflowRatio;
 
+    private boolean allMutationsCompleted = false;
```

Comment:
Did you consider making this `volatile`? In its current usage, it doesn't need it, strictly speaking, because the race is benign and simply results in getting a safer view of the graph.

Reply:
Done